### PR TITLE
PR: Limit sphinx version to `<7.4.0` to prevent bug with splash screen on Windows

### LINKED ISSUE #22353: Splash screen when running from installer shows partially (at least on Windows)
## Problem Description

When launching Spyder installed from the installer the splash screen doesn't show the top image (at least on Windows)

### What steps reproduce the problem?

1. Install Spyder 6.0.0rc1 from the installers
2. Wait for the auto launch after the installation fishised
3. Seems like this also happens when manually launching Spyder with the shortcut

### What is the expected output? What do you see instead?

I think the splash screen should be showing an image, right?

installing_splash

image

### Traceback

Running the command the generated shortcut uses to launch from a cmd (so something like `C:\Users\dalth\AppData\Local\spyder-6\envs\spyder-runtime\python.exe C:\Users\dalth\AppData\Local\spyder-6\envs\spyder-runtime\Scripts\spyder-script.py`) I can get see some output that could be related:

```python-traceback
5hBYx4ZUDAyBfcHpPhCQL GUJp8zFbSyxbUFE1l4Jb5lg1DULM2FVArl8Ro17gDDqBnFsUvOYyBKXqUAc94gNgzHOK/qKGoErM a1copYzKAFnXbAxuneOZnlS+JWepzBxVTucfg1Kjx+H8Efxx/wDge4wZub/Dr/8ABGauXRcwuE57 Ibj2TFgsP7lWkNTWosV5I6EB02kpdg7lg3qIjVsCwanmcs5ieZUpKB0whMqmRWZeblnaBa1NyZVW pZWLnQQGCWSk+JXxW0wlC82iKpxOdMR/hJfR5OZBXiazTcfAzq4ToHd6ZWYHe0VUyVdxgNOUajkd 8d7HvuPFJzhqU107IrIL5IKbA8SnqjWSJgcXqENY7jIdkf8A6S8ohe3xMIkXhIF6j1BFpUw1FRTf cGxTApavxGUNhrBF0GR5hKH7gQxZ1URXELCUaGIyF0wjWDuXGF3Ac6qD1XUKuD4iMpZAd6jdQ3E4 nMauYNkjUtiEOJbuW4wPKviUUGIFtWOggcRQtWPqBOELBPiHaviUxMQ1K3M/H/5I5/N5jbDE5/Bc SXBYuIxniV+NQ3K/AzTLvEv6jvxLqKav7meTNkiZ8/he4Y3AMbpjGjfMVtl6hYSPi3HQLzLKd3LN xDwgjeWMsNxZmUFRlcpEsALFMPxG0OMw+IA1UWKsOtkvgJzUx0s+GE9z2lDgvDzLhIwKYiZI2LMR lINhKQmelqfroh0xG6hkHXuG6HuOSQiCylcTC2fEAWgh3VDTR1DEWt//ACY1whRmJawyLmpQF3jm V2jfERA2Mti3P6iCKv3L4JAIC65Y+b8YlsXbzCKKlwRqOxyHUGCB4RWByQwpmEBNxXKz4lBhPcKt 1L7XT1GxwltnJEMlWRgJjqFgAOqhOcn6jqdozqrXxKtlP1BuwXAViBVR9S5nR6Z1MahyGZY3MG5i 3L1uX7l+5ZNy/ct3L9xa3L9y/ct3PNL9y/cv3B9y/cv3L9zzTyQs3PJL9y/c8080bNzyTyTyTyTz /qXG55IWbldX6SwP0I3b9S93MG55p5oGaG4wDw6RAlG+pXbSuiKpmVFU+JUf6kRcmeopY/qYP8T/ AMhEsKcwyZmlcuplMU/UldmzZf6lNXBEwcYyK8ESSq/+MOOcTibe4rCaxJz5RXQmWKtUrqa2uSKr pXxJmN8SL2lNpBB37kuhlIhiUIjkcWaseP8AFFNnVddTI0Mf648gzVvZmusymuZJeXeSiwLH/wAq gV1tBZgIacbgcEheIvTSAmMgtDj+qCqYDyKkbTwkJU49ShcpSYNIlfSIMoyu0VITKILNorGaxXJp UTb94nM4Bu2X74pHeR1O9SBt8pLcS9IoLviBonSRxrI3aSf/2Q=="
link #image0 is undefined!
```

## Versions

* Spyder version: 6.0.0rc1  (standalone)
* Python version: 3.11.9 64-bit
* Qt version: 5.15.8
* PyQt5 version: 5.15.9
* Operating System: Windows-10-10.0.19045-SP0

### PR DESCRIPTION
**Pull request by mrclary**
## Description of Changes

Limit `sphinx` version to <7.4.0. <=7.4.0 causes issue with Spyder's splash screen.

Also, only use `matplotlib-base` for installer runtime environment. This prevents installing unnecessary `pyside` and `qt6-main` packages.

### Issue(s) Resolved

Fixes #22353

### CODE REVIEW

**Comment by ccordoba12** [comment]
@mrclary, I think  we should include this fix in rc2 to tests the installers before the final release. Do you agree?

**Comment by mrclary** [comment]
Yes. Do you want me to include the changes to remove `matplotlib`?

**Comment by ccordoba12** [comment]
> Yes. Do you want me to include the changes to remove matplotlib?

Ok, please do that too.

File: spyder/plugins/help/confpage.py [hunk 1]
```diff
@@ -43,6 +43,7 @@ def setup_page(self):
         features_group = QGroupBox(_("Additional features"))
         math_box = self.create_checkbox(_("Render mathematical equations"),
                                         'math')
+        # ??? Do we want to increase minimum sphinx requirement for Spyder?
```
Comment (ccordoba12) [inline spyder/plugins/help/confpage.py:46]: Yeah, this is a good idea. Perhaps increasing it to Sphinx 4 or 5 would be better, but I don't know enough about it to take an informed decision.

**Review by ccordoba12 — APPROVED** [review]
Looks good to me, thanks @mrclary!

Note: The size of the Windows and Linux installers decreased in almost 100Mb by using `matplotlib-base`!

File: spyder/plugins/help/confpage.py (hunk 1 above)
Comment (mrclary) [reply spyder/plugins/help/confpage.py:46]: I don't know enough either. It may be best to leave the lower limit where it currently is, at least for now.

**Review by dalthviz — APPROVED** [review]
Thanks @mrclary ! I gave this PR Windows installer a local check and seems like the splash screen is working as expected :+1:

If understood correctly, the sphinx dependency version constraint is only required for Windows right? If so, could it be worthy to only limit the Sphinx version on Windows? Also, could it be worthy to create a follow up issue to hopefully remove the upper constraint when the specific reason why this is happening is discovered, right @ccordoba12 @mrclary ?

**Comment by ccordoba12** [comment]
> If understood correctly, the sphinx dependency version constraint is only required for Windows right? If so, could it be worthy to only limit the Sphinx version on Windows?

I think it's easier to leave that constraint for all OSes for now. Then in 6.0.1 we can try to commit a proper fix.

> Also, could it be worthy to create a follow up issue to hopefully remove the upper constraint when the specific reason why this is happening is discovered, right @ccordoba12 @mrclary ?

Good idea. Could you create one and tag it for 6.0.1? Thanks!

**Comment by dalthviz** [comment]
Created #22374. Merging this to proceed with the release

**Pull request merged**